plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,581 @@
+    end
+    return sub_item_table
+end
+
+function HotKeyShortcuts:attachNewTableToExistingTable(orig_table, second_table)
+    for _, v in ipairs(second_table) do
+        table.insert(orig_table, v)
+    end
+end
+
+--[[
+    This function configures and adds various hotkey shortcuts to the main menu based on the device's capabilities
+    and user settings. It supports different sets of keys for devices with and without keyboards.
+
+    The function performs the following steps:
+    1. Defines sets of cursor keys, page-turn buttons, and function keys.
+    2. Adds the "press" key to function keys if the corresponding setting is enabled.
+    3. If the device has a keyboard, additional sets of keys (cursor, page-turn, and function keys) are appended.
+    4. Adds a menu item for enabling/disabling the use of the press key for shortcuts.
+    5. Adds a menu item for configuring keyboard shortcuts, including cursor keys, page-turn buttons, and function keys.
+    6. If the device has a keyboard, an additional menu item for alphabet keys is added.
+--]]
+function HotKeyShortcuts:addToMainMenu(menu_items)
+    local cursor_keys = {
+        "modifier_plus_up",
+        "modifier_plus_down",
+        "modifier_plus_left",
+        "modifier_plus_right",
+    }
+    local pg_turn = {
+        "modifier_plus_left_page_back",
+        "modifier_plus_left_page_forward",
+        "modifier_plus_right_page_back",
+        "modifier_plus_right_page_forward",
+    }
+    local fn_keys = {
+        "modifier_plus_back",
+        "modifier_plus_home",
+        "modifier_plus_press"
+    }
+    if G_reader_settings:isTrue("press_key_does_hotkeyshortcuts") then
+        table.insert(fn_keys, 1, "press")
+    end
+    if Device:hasKeyboard() then
+        local cursor_keys_haskeyboard = {
+            "alt_plus_up",
+            "alt_plus_down",
+            "alt_plus_left",
+            "alt_plus_right",
+        }
+        self:attachNewTableToExistingTable(cursor_keys, cursor_keys_haskeyboard)
+        local pg_turn_haskeyboard = {
+            "alt_plus_left_page_back",
+            "alt_plus_left_page_forward",
+            "alt_plus_right_page_back",
+            "alt_plus_right_page_forward",
+        }
+        self:attachNewTableToExistingTable(pg_turn, pg_turn_haskeyboard)
+        local fn_keys_haskeyboard = {
+            "modifier_plus_menu",
+            "alt_plus_back",
+            "alt_plus_home",
+            "alt_plus_press",
+            "alt_plus_menu"
+        }
+        self:attachNewTableToExistingTable(fn_keys, fn_keys_haskeyboard)
+    end
+    menu_items.button_press_does_hotkeyshortcuts = {
+        sorting_hint = "physical_buttons_setup",
+        text = _("Use the press key for shortcuts"),
+        checked_func = function()
+            return G_reader_settings:isTrue("press_key_does_hotkeyshortcuts")
+        end,
+        callback = function()
+            G_reader_settings:flipNilOrFalse("press_key_does_hotkeyshortcuts")
```

Comment:
Does this `press_key_does_hotkeyshortcuts` already exists ? Is it used by core ?
If not, and you're adding it, can't it be stored in your `hotkeyshortcuts.lua`, so all is in one place?

Reply:
it doesn't exist elsewhere, it is crreated here and used here only.

Reply:
why can't i have something nice though?

Reply:
(Because _you just haven't earned it yet, baby_.)
Seriously: what ?

Reply:
can't we just leave it with the `G_settings`? they have some too... (points at gesture plugin).

Reply:
You can if you really don't want to bother (but you won't have something "nice").

Reply:
> can't we just leave it with the `G_settings`? they have some too... (points at gesture plugin).
> 

Gestures had them before it was a plugin ;)
It used to be part of core, but that was a limiting pain so i migrated it to a plugin.

Reply:
It's just a matter of storing it as:
```lua
self.hotkeyshortcuts.press_key_does_hotkeyshortcuts
-- which ends up stored as:
self.settings_data.data["press_key_does_hotkeyshortcuts"]
```

Reply:
> You can if you really don't want to bother (but you won't have something "nice").

Mmm… didn’t you say it was fine?

Reply:
I guess this PR will wait for after next release, so you still have time to try making something nice :)

Reply:
But it is already nice.

Reply:
You indeed made it nice by making it a standalone plugin.
The only non-standalone thingy left is the use of G_reader_settings for that single internal setting. So, there's room to make it even nicer.